src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -74,6 +77,34 @@ boot_compinfo_init(spdid_t spdid, captblcap_t *ct, pgtblcap_t *pt, u32_t vaddr)
 }
 
 static void
+boot_newcomp_sinv_alloc(spdid_t spdid, spdid_t intr_spdid)
+{
+		sinvcap_t   sinv;
+		struct cos_compinfo * newcomp_compinfo = new_comp_cap_info[spdid].compinfo; 
+		struct cos_compinfo * interface_compinfo = new_comp_cap_info[intr_spdid].compinfo;
+
+		/* 
+		 * Loop through all undefined symbs 
+		 * TODO: currently hardcoded to 5, fix this. See new_comp_cap_info struct
+		 */	
+		int i = 0;
+		for (i = 0 ; i < 5 ; i++ ) {
+			if ( new_comp_cap_info[spdid].ST_user_caps[i].service_entry_inst > 0) {
+				/* Create sinv capability from ping to pong */
+				sinv = cos_sinv_alloc(newcomp_compinfo, interface_compinfo->comp_cap, (vaddr_t)new_comp_cap_info[spdid].ST_user_caps[i].service_entry_inst);
+				assert(sinv > 0);
+				
+				new_comp_cap_info[spdid].ST_user_caps[i].cap_no = sinv;
+				
+				/* Now that we have the sinv allocated, we can copy in the symb user cap to correct index */
+				memcpy((void *) (new_comp_cap_info[spdid].vaddr_mapped_in_booter + (new_comp_cap_info[spdid].vaddr_user_caps - new_comp_cap_info[spdid].addr_start) + sizeof(struct usr_inv_cap)) , &new_comp_cap_info[spdid].ST_user_caps[i], sizeof(struct usr_inv_cap));
```

Comment:
:+1: also very tricky

Reply:
This looks like it only will map in a *single* capability, and always in the same ucap tbl entry.  Is that correct?

Did you test with multiple functions?